### DISCUSSION TITLE
Slightly OT: Running openWakeWord Jupyter Notebook locally

### DISCUSSION THREAD

**Discussion by mkammes** [dscripka/openWakeWord#166] [Q&A]
Has anyone tried this? I've got some GPU cycles to kill.

I can certainly localize the ipynb file, build an environment with Python 3.6+, and install all the prereqs, then Jupyter, but the pre-reqs needed for the openwakeword notebook differ. I've tried various configs and versions with no luck, including manually adding additional packages.

Timeouts using Google Collab are very common once I start to raise the number of training steps, etc.

Thanks!

**Comment by dscripka** [comment]
This should certainly be possible, as I run the training notebook locally on many different environments. In fact, the environment configuration that is at the beginning of the notebook should be all that is required.

Is there a particular error that you are getting when you try to setup the environment?

**Reply by mkammes** [reply]
Win 11 Pro.

I'm creating and managing the environment using Anaconda Navigator.

My steps:
Create a new Environment, with Python 3.11.9.
I install the Jupyter meta-package (1.0.0), which also installs its 112 dependencies.
Download automatic_model_training_simple.ipynb.
Launch the environment via Terminal.
Activate the Environment, install dependencies: pip install ipython pathlib tqdm datasets numpy scipy torch pyyaml (some are already satisfied)
Open the Environment via Jupyter Notebook. Navigate to automatic_model_training_simple.ipynb.

Errors:
```
Cloning into 'piper-sample-generator'...
'wget' is not recognized as an internal or external command,
operable program or batch file.
ERROR: Could not find a version that satisfies the requirement piper-phonemize (from versions: none)
ERROR: No matching distribution found for piper-phonemize
Collecting webrtcvad
  Using cached webrtcvad-2.0.10-cp311-cp311-win_amd64.whl
Installing collected packages: webrtcvad
Successfully installed webrtcvad-2.0.10
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[1], line 33
     30     if "piper-sample-generator/" not in sys.path:
     31         sys.path.append("piper-sample-generator/")
---> 33     from generate_samples import generate_samples
     35 target_word = 'hey_jinks' # @param {type:"string"}
     37 def text_to_speech(text):

File ~\Downloads\piper-sample-generator\generate_samples.py:14
     12 import numpy as np
     13 import torch
---> 14 import torchaudio
     15 import webrtcvad
     16 from piper_phonemize import phonemize_espeak

ModuleNotFoundError: No module named 'torchaudio'
```
I then install:

- torchaudio (reported as successful)
- wget (reported as successful)
- git clone https://github.com/rhasspy/piper-sample-generator (successful)
- wget -O piper-sample-generator/models/en_US-libritts_r-medium.pt 'https://github.com/rhasspy/piper-sample-generator/releases/download/v2.0.0/en_US-libritts_r-medium.pt'
_'wget' is not recognized as an internal or external command_

So, I decide to try curl, and error:
`curl: (35) schannel: next InitializeSecurityContext failed: CRYPT_E_NO_REVOCATION_CHECK (0x80092012) - The revocation function was unable to check revocation for the certificate.`

I've also attempted to install these dependencies:
```
pip install piper-phonemize (ERROR: Could not find a version that satisfies the requirement piper-phonemize (from versions: none)
ERROR: No matching distribution found for piper-phonemize
pip install webrtcvad(successful)
```

I started all over again and manually downloaded the piper-sample-generator repo.  Error:
```
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
Cell In[5], line 46
     37 def text_to_speech(text):
     38     generate_samples(text = text,
     39                 max_samples=1,
     40                 length_scales=[1.1],
   (...)
     43                 file_names=["test_generation.wav"]
     44                 )
---> 46 text_to_speech(target_word)
     47 Audio("test_generation.wav", autoplay=True)

Cell In[5], line 38, in text_to_speech(text)
     37 def text_to_speech(text):
---> 38     generate_samples(text = text,
     39                 max_samples=1,
     40                 length_scales=[1.1],
     41                 noise_scales=[0.7], noise_scale_ws = [0.7],
     42                 output_dir = './', batch_size=1, auto_reduce_batch_size=True,
     43                 file_names=["test_generation.wav"]
     44                 )

NameError: name 'generate_samples' is not defined
```

So, I've stopped there thus far. I should add that if I leave "kum-puter" as the wakeword, at least the first play works. If I change this to a different word...that's when errors start.

**Reply by dscripka** [reply]
It looks like the main issues is that the `piper-phonemize` package is not available in windows, along with `wget` not working properly (though you can just manually download that model file). You can try manually install the `piper-phonemize` package, but it may not build properly on windows.

Are you able to use WSL? https://learn.microsoft.com/en-us/windows/wsl/install

That might be a better way to setup the environment.
